mephisto/client/review_server.py

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from flask import Flask, Blueprint, send_file, jsonify, request
+import os
+import atexit
+import signal
+import csv
+import sys
+import time
+import threading
+
+
+if False:  # sys.stdout.isatty():
+    pass
+else:
+    # disable all logging out to stdout
+    import logging
+
+    log = logging.getLogger("werkzeug")
+    log.setLevel(logging.ERROR)
+    cli = sys.modules["flask.cli"]
+    cli.show_server_banner = lambda *x: None
+
+
+def run(build_dir):
+    global index_file, app
+    global ready_for_next, current_data, finished, index_file
+    global counter
+    app = Flask(
+        __name__,
+        root_path=os.getcwd(),
+        static_url_path="/static",
+        static_folder=build_dir + "/static",
+    )
+
+    def consume_data():
+        global ready_for_next, current_data, finished, counter
+        data_source = csv.reader(iter(sys.stdin.readline, ""))
+
+        finished = False
+        counter = 0
+        for row in data_source:
+            ready_for_next = threading.Event()
+            current_data = row
+            counter += 1
+            ready_for_next.wait()
+        finished = True
```

Comment:
In future iterations it'll be useful to specify data in/out types, but for now this is a great start for that.

Reply:
Yup! This is how I'm hoping to be inclusive towards use-cases like ego4d.

100% agreed that it would be nice to support more than just csv. I plan to add support for more I/O readers as this evolves. There are also some existing handy utils that allow these transformations, so you can pipe your datasource through a converter util & convert to csv as a stopgap measure. But for now I think this should be good to get us started, with maybe JSON just around the corner.